### PR TITLE
Feat: Add ability to tag Lambda request and response payloads for 2.7…

### PR DESCRIPTION
**Pull request by astuyve**
… and 3.x

### What does this PR do?

When `DD_CAPTURE_LAMBDA_PAYLOAD` is enabled (and tracing is enabled), will automatically wrap and (deeply) tag both the incoming request and outgoing response.

### Motivation

Feature request

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [x] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [x] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: tests/test_tag_object.py
Comment (nhinsch) [inline tests/test_tag_object.py:14]: I would suggest adding test coverage for redaction here

File: datadog_lambda/tag_object.py
Comment (nhinsch) [inline datadog_lambda/tag_object.py:9]: How did we decide on these four keys for redaction? Should we provide a way to redact other keys?

File: CONTRIBUTING.md
Comment (nhinsch) [inline CONTRIBUTING.md:1]: Thank you for cleaning this up! 🙏

File: tests/test_tag_object.py
Comment (astuyve) [reply tests/test_tag_object.py:14]: Good call

File: datadog_lambda/wrapper.py
Comment (nhinsch) [inline datadog_lambda/wrapper.py:186]: Can we enable this feature in the integration tests as well for improved test coverage?

**Comment by nhinsch** [comment]
Super nit: I don't think you need to specify "for 2.7 and 3.x" in the PR title -- all PRs apply to those versions 🙂

(Unless we wait until Thursday to merge this, at which time we will probably be dropping support for 2.7!)

File: datadog_lambda/tag_object.py
Comment (nhinsch) [inline datadog_lambda/tag_object.py:44]: Nit: Why does `_readct_val` begin with an underscore but `shouldTryString` doesn't?

Also, this function name should be `snake_cased` per Python convention.

File: datadog_lambda/tag_object.py
Comment (nhinsch) [inline datadog_lambda/tag_object.py:59]: Ideally it would be nice to add unit tests for the different cases of `shouldTryString` and `_redact_val` as well

File: datadog_lambda/tag_object.py
Comment (astuyve) [reply datadog_lambda/tag_object.py:44]: Because this is my first python PR and I'm a noob :)

Thanks, good catch

File: datadog_lambda/tag_object.py
Comment (astuyve) [reply datadog_lambda/tag_object.py:9]: We provide `datadog.yaml` obfuscation rules - these were 4 common items we wanted to automatically strip out to prevent accidental ingestion of secrets before a user has time to redact for themselves

File: tests/test_tag_object.py
Comment (astuyve) [reply tests/test_tag_object.py]: Done

File: datadog_lambda/tag_object.py
Comment (astuyve) [reply datadog_lambda/tag_object.py:59]: Done!

File: datadog_lambda/wrapper.py
Comment (astuyve) [reply datadog_lambda/wrapper.py:186]: It appears that we don't care about span tags in our integration tests, because the integration tests are unaffected by this ENV variable (off or on). We can do this during an engineering sprint, perhaps?

Comment (nhinsch) [reply datadog_lambda/wrapper.py:186]: Hmm, that's weird, the integration tests are supposed to capture span tags. Let's dig into this a bit more before merging to make sure nothing is wrong. Happy to chat on Zoom about this if you want.

**Review by nhinsch — APPROVED** [review]

**Pull request merged**
